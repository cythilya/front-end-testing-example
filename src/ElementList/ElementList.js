const ElementList = () => {
  return (
    <>
      <div data-test-id="visible-element" style={{ display: 'block' }}>
        這是顯示的元素
      </div>
      <div data-test-id="invisible-element" style={{ display: 'none' }}>
        這是隱藏的元素
      </div>
    </>
  );
};

export default ElementList;
