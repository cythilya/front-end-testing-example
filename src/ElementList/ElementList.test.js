import { render } from '@testing-library/react';
import ElementList from './ElementList';

describe('ElementList', () => {
  it('should display the elements correctly', () => {
    const { getByTestId, queryAllByTestId } = render(<ElementList />);

    // 檢查顯示的元素是否存在
    expect(getByTestId('visible-element')).toBeVisible();
    expect(getByTestId('visible-element')).toBeInTheDocument();
    expect(queryAllByTestId('visible-element')).toHaveLength(1);
    expect(getByTestId('visible-element')).toMatchSnapshot();

    // 檢查隱藏的元素是否存在
    expect(getByTestId('invisible-element')).not.toBeVisible();
    expect(getByTestId('invisible-element')).toBeInTheDocument();
    expect(queryAllByTestId('invisible-element')).toHaveLength(1);
    expect(getByTestId('invisible-element')).toMatchSnapshot();
  });
});
