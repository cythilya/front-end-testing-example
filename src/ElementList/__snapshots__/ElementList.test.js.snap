// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementList should display the elements correctly 1`] = `
<div
  data-test-id="visible-element"
  style="display: block;"
>
  這是顯示的元素
</div>
`;

exports[`ElementList should display the elements correctly 2`] = `
<div
  data-test-id="invisible-element"
  style="display: none;"
>
  這是隱藏的元素
</div>
`;
